### ISSUE TITLE
Different upstream and downstream learning rates

### ISSUE THREAD

**Issue by xt0r3-Cambridge** [s3prl/s3prl#521]
# Feature request

**Is your feature request related to a problem? Please describe.**
I am trying to fine-tune a model with the `--upstream-trainable` flag, but it seems like there is no way to make the upstream model train with a lower learning rate compared to the downstream expert.

**Describe the solution you'd like**
In the optimizer settings, we should have an "upstream_lr" flag. If this is set to a value, then when defining the [optimizer parameters](https://github.com/s3prl/s3prl/blob/c12f170c35626055f6eb2b671105f30b20132bdb/s3prl/optimizers.py#L30C21-L30C21), I would save the upstream layers with their own learning rate, as shown in the [per-parameter options](https://pytorch.org/docs/0.3.0/optim.html#per-parameter-options) of the torch optimizers docs.
 
**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by raotnameh** [comment]
This is so needed.
